packages/components/Notification/src/Notification.tsx

```diff
@@ -34,17 +34,30 @@ export const Notification = compose<NotificationType>({
     action: Button,
   },
   useRender: (userProps: NotificationProps, useSlots: UseSlots<NotificationType>) => {
-    const notificationProps = useNotification(userProps);
     const Slots = useSlots(userProps, (layer) => notificationLookup(layer, userProps));
+    let marginHorizontal = 16;
+    let alignSelf: FlexAlignType = 'flex-start';
+
+    if (['primaryOutlineBar', 'primaryBar', 'neutralBar'].includes(userProps.variant)) {
+      marginHorizontal = 0;
+      alignSelf = 'center' as FlexAlignType;
+    }
+
+    let rootStyle = useMemo(() => {
+      return { variant: userProps.variant, marginHorizontal: marginHorizontal };
+    }, ['variant', 'marginHorizontal']);
+    let messageStyle = useMemo(() => {
+      return { alignSelf: alignSelf };
+    }, ['alignSelf']);
```

Comment:
So that we have more const variables,

```suggestion
    const isBar = (['primaryOutlineBar', 'primaryBar', 'neutralBar'].includes(userProps.variant))
    const marginHorizontal = isBar ? 0: 16
    const alignSelf = isBar ? 'center' : 'flex-start';

    const rootStyle: ViewStyle = useMemo(() => {
      return { style: { marginHorizontal: marginHorizontal } };
    }, ['variant', 'marginHorizontal']);
    let messageStyle: ViewStyle = useMemo(() => {
      return { style: {alignSelf: alignSelf } };
    }, ['alignSelf']);
```

Reply:
Or better yet:

```suggestion
    const isBar = (['primaryOutlineBar', 'primaryBar', 'neutralBar'].includes(userProps.variant))
    const rootStyle: ViewStyle = useMemo(() => {
      const marginHorizontal = isBar ? 0: 16
      return { style: { marginHorizontal: marginHorizontal } };
    }, ['isBar', ]);
    let messageStyle: ViewStyle = useMemo(() => {
      const alignSelf = isBar ? 'center' : 'flex-start';
      return { style: {alignSelf: alignSelf } };
    }, ['isBar']);
```

Reply:
Had to change a couple of things (variant has to be in rootStyle/rootStyle can't be of type ViewStyle for mergeProps to work) to resolve errors so lmk if there's a change here you want to keep

Reply:
Resolved offline